src/main/java/com/cloudogu/scm/ci/cistatus/service/CIStatusService.java

```diff
@@ -59,13 +59,17 @@ public void put(CIStatusStore store, Repository repository, String id, CIStatus
     ciStatusCollection.put(ciStatus);
     getStore(store, repository).put(id, ciStatusCollection);
   }
-
   public CIStatusCollection get(CIStatusStore storeName, Repository repository, String id) {
     PermissionCheck.checkRead(repository);
     CIStatusCollection collection = getStore(storeName, repository).get(id);
     return  collection != null ? collection : new CIStatusCollection();
   }
 
+  public void overwriteCollection(CIStatusStore store, Repository repository, String id, CIStatusCollection collection) {
+    PermissionCheck.checkWrite(repository);
```

Comment:
A user has somehow managed to close a pull request, what triggered this call. Is it wise to check for this permission, here?

Reply:
Maybe not.